src/kw_config_loader.sh

```diff
@@ -141,8 +141,22 @@ function parse_configuration()
 # higher level setting definitions to overwrite lower level ones.
 function load_configuration()
 {
+  local -a config_dirs
+  local IFS=:
+  read -ra config_dirs <<< "${XDG_CONFIG_DIRS:-"/etc/xdg"}"
+  unset IFS
+
   parse_configuration "$KW_ETC_DIR/$CONFIG_FILENAME"
-  parse_configuration "$HOME/.kw/$CONFIG_FILENAME"
+
+  # XDG_CONFIG_DIRS is a colon-separated list of directories for config
+  # files to be searched, in order of preference. Since this function
+  # reads config files in a reversed order of preference, we must
+  # traverse it from back to top
+  for ((i = ${#config_dirs[@]} - 1; i >= 0; i--)); do
+    parse_configuration "${config_dirs["$i"]}/$KWORKFLOW/$CONFIG_FILENAME"
+  done
+
+  parse_configuration "${XDG_CONFIG_HOME:-"$HOME/.config"}/$KWORKFLOW/$CONFIG_FILENAME"
```

Comment:
fwiu from the `setup.sh`, we will move `$HOME/.config` to the new folder. If I am correct, do we really need this `${XDG_CONFIG_HOME:-"$HOME/.config"}`?

Reply:
Actually, $HOME/.config is the new folder!